### PR TITLE
Feature/us12482 hubspot kc and student ministry data sync from mp to hubspot derived information

### PR DESCRIPTION
**Pull request by agause**
@jpc11623,

This enables Kid's Club and Student Ministry age/grade count data to be synced to HubSpot for any active, registered contact associated with a household containing children in the relevant KC/SM age/grade range.

### CODE REVIEW

File: Crossroads.Service.HubSpot.Sync.App/Crossroads.Service.HubSpot.Sync.App.csproj [hunk 1]
```diff
@@ -1,12 +1,12 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <ItemGroup>
-    <PackageReference Include="Crossroads.Web.Common" Version="1.1.13" />
+    <PackageReference Include="Crossroads.Web.Common" Version="2.0.2" />
```
Comment (jpc11623) [inline Crossroads.Service.HubSpot.Sync.App/Crossroads.Service.HubSpot.Sync.App.csproj]: Bump web common

File: CI/SQL/20180502_111897_US12482-cr-ChildAgeAndGradeCountsByHousehold-table.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
+use MinistryPlatform
```
[the remark this answers is not in the document]
Comment (agause) [reply CI/SQL/20180502_111897_US12482-cr-ChildAgeAndGradeCountsByHousehold-table.sql]: @jpc11623 Good question. We persist in order to limit the traffic from MP to HubSpot by sending only deltas after the initial load.

HubSpot has a 10/sec and 40K/day request rate limit. So we never get anywhere near it, I'm trying to be conservative and limit the chatter. Who knows how else we'll integrate with it in the future.

File: Crossroads.Service.HubSpot.Sync.ApplicationServices.Tests/Services/CleanUpSyncActivityTests.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+using Xunit;
+
+namespace Crossroads.Service.HubSpot.Sync.ApplicationServices.Test.Services
+{
+    public class CleanUpSyncActivityTests
+    {
+        private readonly CleanUpSyncActivity _fixture = new CleanUpSyncActivity();
+
+        [Fact]
+        public void given_a_null_activity_clean_up_should_not_throw_an_exception()
```
[the remark this answers is not in the document]
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.ApplicationServices.Tests/Services/CleanUpSyncActivityTests.cs]: @jpc11623 I'll fix all the unit test naming things.

File: Crossroads.Service.HubSpot.Sync.Data/Crossroads.Service.HubSpot.Sync.Data.csproj [hunk 4]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Crossroads.Web.Common" Version="1.1.13" />
+    <PackageReference Include="Crossroads.Web.Common" Version="2.0.2" />
```
[the remark this answers is not in the document]
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.Data/Crossroads.Service.HubSpot.Sync.Data.csproj]: @jpc11623 Will do.

File: Crossroads.Service.HubSpot.Sync.Data/HubSpot/Models/Request/IContact.cs [hunk 5]
```diff
@@ -5,6 +5,8 @@ namespace Crossroads.Service.HubSpot.Sync.Data.HubSpot.Models.Request
 {
     public interface IContact
     {
+        string Email { get; set; }
```
[the remark this answers is not in the document]
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.Data/HubSpot/Models/Request/IContact.cs]: @jpc11623 Yes, the JSON annotation varies by implementation. BulkContact is req'd, whereas SerialContact mustn't have it (or HubSpot will throw a hissy fit).

File: Crossroads.Service.HubSpot.Sync.App/Crossroads.Service.HubSpot.Sync.App.csproj (hunk 1 above)
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.App/Crossroads.Service.HubSpot.Sync.App.csproj]: @jpc11623 Will do.

File: Crossroads.Service.HubSpot.Sync.Core/Utilities/Util.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        /// swallow the exception. Useful for wrapping around methods whose failure should not
+        /// preclude subsequent processing. Any logging should have been done higher up in the
+        /// stack.
+        /// </summary>
+        public static void TryCatchSwallow(Action methodToMuzzle, Action catchMethod = null, Action finallyMethod = null)
+        {
+            try { methodToMuzzle(); } catch { catchMethod?.Invoke(); } finally { finallyMethod?.Invoke(); }
+        }
+
+        /*
```
[the remark this answers is not in the document]
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.Core/Utilities/Util.cs]: @jpc11623 Leaving this for now. I'll comment on its future purpose: the hope is for async, rate-limited requests.

File: Crossroads.Service.HubSpot.Sync.Data/Crossroads.Service.HubSpot.Sync.Data.csproj (hunk 4 above)
[the remark this answers is not in the document]
Comment (agause) [reply Crossroads.Service.HubSpot.Sync.Data/Crossroads.Service.HubSpot.Sync.Data.csproj]: @jpc11623 appears to be a v2.0.6 as of today. I guess I'll grab that one instead.

**Pull request merged**